### PR TITLE
Enable strict mode in storybook on a per story basis

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: .storybook/strictmode.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ) : children;
+
+  return (
+    <Provider theme={defaultTheme}>
+      <Checkbox
+        marginTop={10}
+        isSelected={selected}
+        onChange={() => setSelected(selected => !selected)}>
+        Toggle strict mode
+      </Checkbox>
```
Comment (LFDanLu) [inline .storybook/strictmode.js:23]: TODO: still trying to figure out if it is possible to have this be a option in the `controls` panel instead

Comment (snowystinger) [reply .storybook/strictmode.js:23]: Can't put it there, but you could put it up in the toolbar across the top of the stories, where we put the locale/scale/theme

Comment (LFDanLu) [reply .storybook/strictmode.js:23]: urgh, I was hoping that https://storybook.js.org/docs/react/essentials/controls#parameters was alluding to being able to set a control param globally but putting it in the toolbar sounds good as well

Comment (snowystinger) [reply .storybook/strictmode.js:23]: you probably could through a decorator but I think it'd be a lot more work than it's worth, plus you'd have to apply it to every component, there's a number of them with no controls
I think toolbar is the better solution

**Comment by LFDanLu** [comment]
test comment

**Comment by LFDanLu** [comment]
test 2

File: .storybook/custom-addons/strictmode/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    channel.on('strict/updated', updateStrict);
+    return () => {
+      channel.removeListener('strict/updated', updateStrict);
+    };
+  }, []);
+
+  return isStrict ? (
+    <StrictMode>
+      {children}
+    </StrictMode>
```
Comment (LFDanLu) [inline .storybook/custom-addons/strictmode/index.js:23]: some weirdness, StrictMode is being applied when switching stories for some reason...

Comment (LFDanLu) [reply .storybook/custom-addons/strictmode/index.js:23]: nvm, me dumb

File: .storybook/custom-addons/strictmode/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+function StrictModeDecorator(props) {
+  let {children} = props;
+  let [isStrict, setStrict] = useState(getQueryParams()?.strict === 'true' || false);
+
+  useEffect(() => {
+    let channel = addons.getChannel();
+    let updateStrict = (val) => {
+      setStrict(val);
+    };
+    channel.on('strict/updated', updateStrict);
```
Comment (LFDanLu) [inline .storybook/custom-addons/strictmode/index.js:14]: I would've like to used the storybook globals but for some reason it stopped working

**Review by snowystinger — COMMENTED** [review]
Well, the storybook seems to be running just fine. This is somewhat problematic for me though, since I can't seem to do anything that gives a StrictMode error. Have you been able to see any?

**Comment by LFDanLu** [comment]
A good candidate is ComboBox (or anything that uses a overlay I believe).

**Comment by snowystinger** [comment]
> A good candidate is ComboBox (or anything that uses a overlay I believe).

o, I figured it out, can't test it in storybook built by CI, that's production.... haha, have to do it locally

**Review by reidbarber — APPROVED** [review]
Tested locally, LGTM.

**Review by snowystinger — APPROVED** [review]
Great, I can see NumberField not working when strict mode is on and it works when strict mode is off.

**Review by reidbarber — APPROVED** [review]

**Pull request merged**